torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -106,24 +106,28 @@ static std::vector<int64_t> seq_to_aten_shape(PyObject *py_seq) {
 }
 
 PyObject* tensor_to_numpy(const at::Tensor& tensor) {
-  if (!is_numpy_available()) {
-    throw std::runtime_error("Numpy is not available");
-  }
-  if (tensor.device().type() != DeviceType::CPU) {
-    throw TypeError(
+  TORCH_CHECK(is_numpy_available(), "Numpy is not available");
+
+  TORCH_CHECK(tensor.device().type() == DeviceType::CPU,
       "can't convert %s device type tensor to numpy. Use Tensor.cpu() to "
       "copy the tensor to host memory first.", tensor.device().str().c_str());
-  }
-  if (tensor.layout() != Layout::Strided) {
-      throw TypeError(
-        "can't convert %s layout tensor to numpy."
-        "convert the tensor to a strided layout first.", c10::str(tensor.layout()).c_str());
-  }
-  if (at::GradMode::is_enabled() && tensor.requires_grad()) {
-    throw std::runtime_error(
-        "Can't call numpy() on Tensor that requires grad. "
-        "Use tensor.detach().numpy() instead.");
-  }
+
+  TORCH_CHECK(tensor.layout() == Layout::Strided,
```

Comment:
Can you use `TORCH_CHECK_TYPE()` here so that the error type remains the same?
Same for the device above.